File: Maximum-Difference-Between-Adjacent-Elements-in-a-Circular-Array.java
class Solution {
    public int maxAdjacentDistance(int[] nums) {
        int maxd=Math.abs(nums[nums.length-1]-nums[0]);
        for(int i=1;i<nums.length;i++){
            maxd=Math.max(maxd,Math.abs(nums[i]-nums[i-1]));
        }
        return maxd;
    }
}

File: Maximum-Difference-Between-Adjacent-Elements-in-a-Circular-Array.cpp
class Solution {
public:
    int maxAdjacentDistance(vector<int>& nums) {
        int maxd=abs(nums[nums.size()-1]-nums[0]);
        for(int i=1;i<nums.size();i++){
            maxd=max(maxd,abs(nums[i]-nums[i-1]));
        }
        return maxd;
    }
};